rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+      - [Tokens - Easing](#tokens---easing)
+  - [Frameworks/Libraries for motion](#frameworkslibraries-for-motion)
+    - [Web (React)](#web-react)
+    - [Mobile (React Native)](#mobile-react-native)
+- [Drawbacks/Constraints](#drawbacksconstraints)
+    - [Library specific tokens](#library-specific-tokens)
+    - [React Native `v0.62`](#react-native-v062)
+    - [Multi-step keyframe animation with React Native](#multi-step-keyframe-animation-with-react-native)
+- [Alternatives](#alternatives)
+    - [Spring Animations instead of Easing Animations](#spring-animations-instead-of-easing-animations)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+# Summary
+
+
+# Motivation
+### Why are we doing this?
+Motion brings your UI to life. It makes your UI feel closer to our physical reality. Just like we have **pre-defined physical laws** in the **real world** that dictate the reactions to our different actions (like throwing a ball high up and expecting it to fall back down in a predictable manner), we want to define similar **set of rules** for **motion** **within** our **Design System**. If two balls thrown up with the same force, come back down in the same manner then why should two buttons on a UI react differently when hovered over?
+
+With this RFC, we want to **establish** certain **rules** (*tokens*) through which we can build predictable motion for our components and **evaluate** **multiple ways** through which we can **use these rules** to build motion within Blade.
+
+### What use cases does it support?
+The rules we establish for motion here can be applied for both Web (**React**) and Mobile (**React Native**) 
+
+### What is the expected outcome?
+1. Have a **set of tokens** that can be consumed within our components for animating them
+2. Conclude on **how we can consume** these **tokens** to create animations
+
+
+# Detailed Design
+## Types of Motion
+We can have 2 types of motion while building user interfaces,
+1. **Realtime**: User is directly interacting with an object on the UI.
+
+    *Example: When a user drags a carousel, the carousel slides as per the user's drag.*
+    
+    <img alt="Realtime Animation Example" src="./images/motion/realtime.gif" width="300px">
+
+2. **Non-Realtime**: The object's behavior is *post-interactive* i.e it occurs after a user's action, and is transitional.
+
+    *Example: When a user taps on a carousel card, it flips over within a set timeframe*
+    
+    <img alt="Realtime Animation Example" src="./images/motion/non-realtime.gif" width="300px">
+
+## Scope of Motion
+As of this RFC, we will be defining rules of motion only for **Non-Realtime** interactions. 
+
+Few examples of non-realtime motion are:
+1. Changing the size of a button on mouse hover
+2. Changing the opacity of a card on press
+3. Fading out a modal when the user closes it
+
+
+## Actions of Motion
```

Comment:
Actions of motion can be a combination of 2 like scale and fill.

Reply:
Yeah, it can be combinations as well. I'll add that here.